pybossa/api/task.py

```diff
@@ -110,6 +111,7 @@ def _preprocess_post_data(self, data):
                     data['exported'] = True
             except Exception as e:
                 raise BadRequest('Invalid gold_answers')
+        data["expiration"] = get_task_expiration(data.get('expiration'))
```

Comment:
This change will always set "expiration" to some value and it will never be NULL. So if a task without any file and the project owner wants this task to last longer(e.g. 365 days), will this change affect them?

It looks to me we can keep this change, but in addition, we need to change the implementation of ```get_task_expiration``` something as follows:
```
def get_task_expiration(current_expiration, task):
    if not has_file(task):  # need to implement has_file function
        return current_expiration
    validity = current_app.config.get('REQUEST_FILE_VALIDITY_IN_DAYS', 60)
    return _get_task_expiration(current_expiration, validity)
```

Reply:
Yes, this change will have expiration set always thereby having consistent behavior that each task expires at defined time as configured under settings. Fyi, in order to support file based expiration, we may need additional configuration under project settings defining which fields contains files as under public, user can have files under s3 for which field names could be anything. If required, let's discuss offline.